swift-sdk/NetworkHelper.swift

```diff
@@ -0,0 +1,117 @@
+//
+//
+//  Created by Tapash Majumder on 7/24/18.
+//  Copyright © 2018 Iterable. All rights reserved.
+//
+
+import Foundation
+
+typealias SendRequestValue = [AnyHashable : Any]
+
+struct SendRequestErrorType {
```

Comment:
This is not NetworkError. We have to have both errorMessage and data for OnFailureCallback. Not sure why data is need though.

Reply:
Looking at the struct name, I just thought it was a class specifying the type of the error it encountered (an enum).

Reply:
What does `Type` in the name mean?

Reply:
Just signifies that it is not actually an 'Error' which is a Swift protocol. I suppose we could actually implement the Error protocol and call it SendRequestError instead of SendRequestErrorType.

Reply:
Oh, you're right, I didn't think of `Error` as a separate protocol. But I think having `Type` suffix (as well as `Class`, `Struct`, etc.) as part of the actual name of the type isn't great since it doesn't add to the description the type itself and can be interpreted as something else (like a type/category of an error).
Maybe `NetworkErrorData`? (or implement `Error` if that isn't going to break things)